qa/integration-tests/src/test/java/io/camunda/zeebe/it/client/ClientCancelPendingCommandTest.java

```diff
@@ -43,4 +48,42 @@ void shouldCancelCommandOnFutureCancellation() {
     Awaitility.await("until no long polling clients are waiting")
         .untilAsserted(() -> assertThat(metrics.getBlockedRequestsCount("type")).isZero());
   }
+
+  @Test
+  void shouldRemoveStreamOnCancel() {
+    // given
+    final var uniqueWorkerName = UUID.randomUUID().toString();
+    final var stream =
+        client
+            .newStreamJobsCommand()
+            .jobType("jobs")
+            .consumer(ignored -> {})
+            .workerName(uniqueWorkerName)
+            .send();
+
+    // when
+    awaitStreamRegistered(uniqueWorkerName);
+    stream.cancel(true);
+
+    // then
+    awaitStreamRemoved(uniqueWorkerName);
+  }
+
+  private void awaitStreamRegistered(final String workerName) {
+    final var brokerBridge = CLUSTER.getBrokerBridge(0);
+    final var jobStreamService = brokerBridge.getJobStreamService().orElseThrow();
+    final var assertions = JobStreamServiceAssert.assertThat(jobStreamService);
```

Comment:
:upside_down_face: Any advantage of reusing this object? It reads better as `JobStreamServiceAssert.assertThat(jobStreamService).hasStreamWithWorker(..)` compared to `assertions.hasStreamWithWorker(..)`.

Reply:
Just avoiding to unnecessarily recreate it :shrug: I can change it